govuk-one-login/data-vocab
# refactor(ts-types): TypeScript generator uses combined JSON Schema.

Uses a combined JSON Schema document as the input for TypeScript type generation.

# Background
- The Java type generator uses a combined JSON Schema, as of #101 
- Historically, the TypeScript generator used separate schema files, which led to duplication of types

# What changed
- The TypeScript generator now uses a combined schema file as its input, to align it to the Java generator
- This has allowed further file structure simplifications, following #104 

# Testing it
- Follow the prerequisites in `code-generators/typescript-types/README.md`
- Run `scripts/generate_ts_types.sh`, which
  - produces a combined schema file under `code-generators/typescript-types/schemas`
  - generate types under  `code-generators/typescript-types/dist`
- Test with the example project (`examples/typescript-node`):
  - Replace the dependency in `package.json` with a local dependency: `npm install ../../code-generators/typescript-types/dist`
  - Run the example project (see its readme), which consumes it without any type errors
